pkg/solana/txm/txm.go

```diff
@@ -377,135 +381,219 @@ func (txm *Txm) confirm() {
 		case <-ctx.Done():
 			return
 		case <-tick:
-			// get list of tx signatures to confirm
-			sigs := txm.txs.ListAll()
-
-			// exit switch if not txs to confirm
-			if len(sigs) == 0 {
+			// If no signatures to confirm and rebroadcast, we can break loop as there's nothing to process.
+			if txm.InflightTxs() == 0 {
 				break
 			}
 
-			// get client
 			client, err := txm.client.Get()
 			if err != nil {
-				txm.lggr.Errorw("failed to get client in soltxm.confirm", "error", err)
-				break // exit switch
+				txm.lggr.Errorw("failed to get client in txm.confirm", "error", err)
+				break
 			}
+			txm.processConfirmations(ctx, client)
 
-			// batch sigs no more than MaxSigsToConfirm each
-			sigsBatch, err := utils.BatchSplit(sigs, MaxSigsToConfirm)
-			if err != nil { // this should never happen
-				txm.lggr.Fatalw("failed to batch signatures", "error", err)
-				break // exit switch
+			// In case all txes where confirmed and there's nothing to rebroadcast.
+			// This check saves making 2 RPC calls (slot height + blockhash) when there's nothing to process.
+			if len(txm.txs.ListAllBroadcastedTxs()) == 0 {
```

Comment:
I could call ListAllExpiredBroadcastedTxs(val) with val being really high (max.Int maybe) and it will return all broadcasted txes. Despite being dirty, it would let us remove ListAllBroadcastedTxs() implementation and prevent doing the RPC call to get the slotHeight

Reply:
done